assets/src/edit-story/components/panels/sizePosition/panel.js

```diff
@@ -131,10 +132,30 @@ function SizePositionPanel({
   usePresubmitHandlers(lockAspectRatio, height, width);
 
   const handleSetBackground = useCallback(() => {
-    combineElements({
-      firstId: selectedElements[0].id,
-      secondId: currentBackgroundId,
-    });
+    const setBackground = (baseColor) => {
+      if (!baseColor) {
+        combineElements({
+          firstElement: selectedElements[0],
+          secondId: currentBackgroundId,
+        });
+      } else {
+        combineElements({
+          firstElement: {
+            ...selectedElements[0],
+            resource: {
+              ...selectedElements[0].resource,
+              baseColor,
+            },
+          },
+          secondId: currentBackgroundId,
+        });
+      }
+    };
+    if (selectedElements[0].resource?.baseColor) {
+      setBackground();
+    } else {
+      getMediaBaseColor(selectedElements[0].resource, setBackground);
+    }
```

Comment:
If there's no resource, the `else` branch will be used, which will probably fail. So the  `?` in `selectedElements[0].resource?.baseColor` should probably be removed or handled separately.